server/routes/index.ts

```diff
@@ -883,6 +945,27 @@ export function defineRoutes(router: IRouter) {
   );
 }
 
+/**
+ * A helper function to wrap API calls with the appropriate client. If the multiple datasources feature is disabled, it will use
+ * the existing client provided by the security plugin. Otherwise, it will use the one provided by the datasource plugin based on the id
+ * that we extract via the UI.
+ */
+const wrapRouteWithDataSource = async (
+  dataSourceEnabled: boolean,
+  context: RequestHandlerContext,
+  request: OpenSearchDashboardsRequest<unknown, any, any>,
+  endpoint: string,
+  body?: Record<string, string>
+) => {
+  if (!dataSourceEnabled || !request.query?.dataSourceId) {
+    const client = context.security_plugin.esClient.asScoped(request);
+    return await client.callAsCurrentUser(endpoint, body);
+  } else {
+    const client = context.dataSource.opensearch.legacy.getClient(request.query?.dataSourceId);
```

Comment:
qq, why did we decide to use legacy client here?

Reply:
I was trying to maintain parity with the existing call. The security_plugins's esClient is a legacy client: [code ref](https://github.com/opensearch-project/security-dashboards-plugin/blob/8f0a52d0fba2859c6de7054594fe81a76ae1c7dc/server/plugin.ts#L52). I don't know if it was used for compatibility with older versions/opendistro/elasticsearch, but I think maintaining parity is fine for now. Should we create an issue to investigate this?

Reply:
done. https://github.com/opensearch-project/security-dashboards-plugin/issues/1903